packages/react/src/components/login/client/oidcClient.test.ts

```diff
@@ -0,0 +1,777 @@
+        acr_values: 'acr_values',
+        extraQueryParams: { extraParam1: 'extra' },
+        extraTokenParams: { extraTokenParam1: 'extra' },
+        nonce: 'nonce',
+        state: { stateValue: 1, path: '/applications' },
+      };
+      await waitForLoginToTimeout({ ...loginParams, language: 'sv' });
+      expect(signinRedirect).toHaveBeenNthCalledWith(1, {
+        ...loginParams,
+        extraQueryParams: {
+          ...loginParams.extraQueryParams,
+          ui_locales: 'sv',
+        },
+      });
+    });
+  });
+  describe('.logout()', () => {
+    beforeEach(async () => {
+      testData = await initTests({});
+      const { oidcClient } = testData;
+      setSignInResponse();
+      await oidcClient.handleCallback();
+    });
+    it('should add given language to the logout url', async () => {
+      const { oidcClient, userManager } = testData;
+      const signoutRedirectSpy = jest.spyOn(userManager, 'signoutRedirect');
+      const loginParams = { language: 'sv' };
+      await waitForLogoutToTimeout(loginParams);
+      expect(signoutRedirectSpy).toHaveBeenCalledTimes(1);
+      expect(signoutRedirectSpy).toHaveBeenNthCalledWith(1, {
+        extraQueryParams: {
+          ui_locales: loginParams.language,
+        },
+      });
+      await waitFor(() => {
+        expect(mockedWindowControls.getCallParameters().get('ui_locales')).toBe(loginParams.language);
+      });
+      expect(oidcClient.isAuthenticated()).toBeFalsy();
+    });
+    it('should pass other LogoutProps than "language" to signoutRedirect and convert "language" to an extraQueryParam', async () => {
+      const { userManager } = testData;
+      const signoutRedirectSpy = jest.spyOn(userManager, 'signoutRedirect');
+      const loginParams: LogoutProps = {
+        extraQueryParams: { extraParam1: 'extra' },
+        state: { stateValue: 2, path: '/logout' },
+        id_token_hint: 'id_token_hint',
+        post_logout_redirect_uri: 'post_logout_redirect_uri',
+      };
+      await waitForLogoutToTimeout({ ...loginParams, language: 'sv' });
+      expect(signoutRedirectSpy).toHaveBeenNthCalledWith(1, {
+        ...loginParams,
+        extraQueryParams: {
+          ...loginParams.extraQueryParams,
+          ui_locales: 'sv',
+        },
+      });
+    });
+    it('should remove the user from sessionStorage', async () => {
+      const userManagerProps = getDefaultOidcClientTestProps().userManagerSettings as UserManagerSettings;
+      const userFromStorage = getUserFromStorage(userManagerProps);
+      expect(userFromStorage).not.toBeNull();
+      await waitForLogoutToTimeout();
+      const userFromStorageAfterLogout = getUserFromStorage(userManagerProps);
+      expect(userFromStorageAfterLogout).toBeNull();
+    });
+  });
+  describe('.handleCallback()', () => {
+    it('should return the same user returned from signinRedirectCallback. Except state is converted to an object and userState is removed (by the oidc-client-ts)', async () => {
```

Comment:
I added this same question below: "why test names cannot be long? Best place to tell clearly what is tested and expected." I have never commented what a test does, because it can be described it the test name. 🤔